### PR TITLE
add  security-severity score to code scanning query list

### PR DESCRIPTION
**Pull request by felickz**
Add `security-severity` metadata tag to the [Build code scanning query list
](https://github.com/github/codeql/actions/workflows/query-list.yml)
- Security Score calculation described in blog: [CodeQL code scanning: new severity levels for security alerts](https://github.blog/changelog/2021-07-19-codeql-code-scanning-new-severity-levels-for-security-alerts/)
- [Metadata](https://codeql.github.com/docs/writing-codeql-queries/metadata-for-codeql-queries/) definition for `security-severity`

Ex:
image

Sample from [runs](https://github.com/github/codeql/actions/workflows/query-list.yml?query=actor%3Afelickz+):  
- [code-scanning-query-list.zip](https://github.com/github/codeql/actions/runs/8711772483/artifacts/1419638823) (reran april 16)
- ~[code-scanning-query-list.zip](https://github.com/github/codeql/actions/runs/8662600957/artifacts/1408714031) (reran April 4 2024)~

### CODE REVIEW

**Review by adityasharad — DISMISSED** [review]

**Comment by RasmusWL** [comment]
Let's look at getting this PR merged in the new year :blush:

File: misc/scripts/generate-code-scanning-query-list.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def subprocess_run(cmd):
         csvwriter = csv.writer(sys.stdout)
         csvwriter.writerow([
             "Query filename", "Suite", "Query name", "Query ID",
-            "Kind", "Severity", "Precision", "Tags"
+            "Kind", "Severity", "Security score", "Precision", "Tags"
```
Comment (RasmusWL) [inline misc/scripts/generate-code-scanning-query-list.py:162]:
```suggestion
            "Kind", "Severity", "Precision", "Tags", "Security score"
```

File: misc/scripts/generate-code-scanning-query-list.py [hunk 2]
```diff
@@ -197,6 +197,7 @@ def subprocess_run(cmd):
                         get_query_metadata('id', meta, queryfile_nwo),
                         get_query_metadata('kind', meta, queryfile_nwo),
                         get_query_metadata('problem.severity', meta, queryfile_nwo),
+                        get_query_metadata('security-severity', meta, queryfile_nwo),
                         get_query_metadata('precision', meta, queryfile_nwo),
                         get_query_metadata('tags', meta, queryfile_nwo)
```
Comment (RasmusWL) [inline misc/scripts/generate-code-scanning-query-list.py:202]:
```suggestion
                        get_query_metadata('precision', meta, queryfile_nwo),
                        get_query_metadata('tags', meta, queryfile_nwo),
                        get_query_metadata('security-severity', meta, queryfile_nwo),
```

**Review by RasmusWL — COMMENTED** [review]
I've suggested changes so the new `"Security score"` column is added as the last column. This will make it easier if any script is reading the files using positional indexing.

(I personally don't know of any, but since I don't see any cost to doing this, I think we might as well play nice)

Ohterwise LGTM :+1:

**Comment by felickz** [comment]
> Sample from [runs](https://github.com/github/codeql/actions/workflows/query-list.yml?query=actor%3Afelickz+):
> 
> * [code-scanning-query-list.zip](https://github.com/github/codeql/actions/runs/8711772483/artifacts/1419638823) (reran april 16)

Accepted and retested... looks good.

Sample from [runs](https://github.com/github/codeql/actions/workflows/query-list.yml?query=actor%3Afelickz+):

- [code-scanning-query-list.zip](https://github.com/github/codeql/actions/runs/8711772483/artifacts/1419638823) (reran april 16)

**Pull request merged**